proposals/using.md

```diff
@@ -0,0 +1,164 @@
+# using patterns and locals
```

Comment:
The "using patterns" is confusing terminology. I think it should be "pattern-based using" instead (just like we do "pattern-base foreach", etc).